aws/corehandlers/handlers.go

```diff
@@ -70,6 +71,25 @@ var reStatusCode = regexp.MustCompile(`^(\d{3})`)
 // SendHandler is a request handler to send service request using HTTP client.
 var SendHandler = request.NamedHandler{Name: "core.SendHandler", Fn: func(r *request.Request) {
 	var err error
+
+	// Checking if cancel channel has been closed
+	//
+	// Canceling a request is non-deterministic. If a request has been canceled,
+	// then the transport will choose, randomly, one of the state channels during
+	// reads or getting the connection.
+	//
+	// readLoop() and getConn(req *Request, cm connectMethod)
+	// https://github.com/golang/go/blob/master/src/net/http/transport.go
+	if r.IsCanceled() {
```

Comment:
Instead of making this check. It might be better to just document the expectation, in `aws/request:Request.Send`.  There are many stages in the chain of making a request where a request could of been canceled that this logic would not be able to capture.

I think a more complete fix would be documentation the expectation, and updating the tests to reflect this.